src/functions/cart-reducer.js

```diff
@@ -0,0 +1,54 @@
+
+const cartReducerFunc = (state, action) => {
+    switch (action.type) {
+      case "ADD-TO-CART":
+        return {
+          ...state,
+          cartItems: state.cartItems + 1,
+          cartList: [...state.cartList, {...action.value,cartQuantity:1}],
+          price:state.price + action.value.originalPrice,
+          discount: state.discount + (action.value.originalPrice - action.value.price),
+          totalPrice: state.totalPrice + action.value.price,
```

Comment:
It seems you're using these function many times below as well with just value changes, so you can make a separate function and pass value to change.
Looks clustered now.